### DISCUSSION TITLE
How to pause all queued animations?

### DISCUSSION THREAD

**Discussion by joeshub** [pmndrs/react-spring#1072] [Support]
I'm creating some animations and then trying to stop them all. I'm not sure why but it appears the animations only stop the current one, and not all the queued animations, so that they end up causing issues when used with `react-use-gesture`

This box will animate left, right, and stop at center.

Drag the box during the initial animations. It should pause in place, stop all animations and become draggable.

Click box to pause queued animations.

Issue: Clicking box again before the 3 animations expire causes box to jump.

Here's a codesandbox for this question
https://codesandbox.io/s/goofy-platform-fnx8n?file=/src/App.js

**Comment by aleclarson** [comment]
You probably want to call the `pause` method when dragging starts, but you're currently calling it from an `onClick` handler, which only runs on mouse up.

**Reply by joeshub** [reply]
Ah Thanks! ok getting closer, now calling it inside of the onDrag cb, though it seems to do something other than pause the animation. It seems to go to the last spring value. Is there a way to just have it pause there?

https://codesandbox.io/s/goofy-platform-fnx8n?file=/src/App.js

**Reply by aleclarson** [reply]
I got your sandbox to pause/resume the chain animation on click, but I had to disable the drag animation to do so, because chain animations are cancelled when another animation is declared. There's no easy way to avoid that for now. Unfortunately, you'll need to roll your own pausing/resuming mechanism that remembers where you were in the animation chain before dragging began.

https://codesandbox.io/s/wandering-leftpad-062vj

**Reply by joeshub** [reply]
> chain animations are cancelled when another animation is declared. 

Ah that's good to know! I ended up using the `stop` method and getting what I wanted that way, though not able to pause yet, it works for what I needed. Thanks for your help again

https://codesandbox.io/s/jovial-antonelli-qlx5j
